results/ml_v1/model_parameters.txt

LightGMB: AUS: Detrended Drift
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.01
max_depth: 10
min_child_samples: 39
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 610
num_leaves: 25
random_state: 42
reg_alpha: 0.4
reg_lambda: 0.1
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: AUS: Detrended Drift
objective: reg:squarederror
enable_categorical: False
max_depth: 6
min_child_weight: 11
missing: nan
n_estimators: 950
random_state: 42
reg_alpha: 0.5
reg_lambda: 0.6
eta: 0.09

LightGMB: AUS: Detrended Diffusion
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.14
max_depth: 11
min_child_samples: 48
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 550
num_leaves: 22
random_state: 42
reg_alpha: 0.0004
reg_lambda: 0.9
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: AUS: Detrended Diffusion
objective: reg:squarederror
enable_categorical: False
max_depth: 6
min_child_weight: 24
missing: nan
n_estimators: 510
random_state: 42
reg_alpha: 0.0
reg_lambda: 0.3
eta: 0.42

LightGMB: AUS: Original Drift
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.012
max_depth: 4
min_child_samples: 44
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 405
num_leaves: 38
random_state: 42
reg_alpha: 0.9
reg_lambda: 0.5
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: AUS: Original Drift
objective: reg:squarederror
enable_categorical: False
max_depth: 3
min_child_weight: 2
missing: nan
n_estimators: 324
random_state: 42
reg_alpha: 0.1
reg_lambda: 0.4
eta: 0.01

LightGMB: AUS: Original Diffusion
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.17
max_depth: 11
min_child_samples: 24
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 168
num_leaves: 29
random_state: 42
reg_alpha: 0.001
reg_lambda: 0.4
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: AUS: Original Diffusion
objective: reg:squarederror
enable_categorical: False
max_depth: 6
min_child_weight: 24
missing: nan
n_estimators: 510
random_state: 42
reg_alpha: 0.0005
reg_lambda: 0.3
eta: 0.42

LightGMB: CE: Detrended Drift
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.03
max_depth: 5
min_child_samples: 75
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 873
num_leaves: 14
random_state: 42
reg_alpha: 0.1
reg_lambda: 1.0
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: CE: Detrended Drift
objective: reg:squarederror
enable_categorical: False
max_depth: 7
min_child_weight: 20
missing: nan
n_estimators: 381
random_state: 42
reg_alpha: 0.0
reg_lambda: 0.1
eta: 0.04

LightGMB: CE: Detrended Diffusion
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.14
max_depth: 11
min_child_samples: 48
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 550
num_leaves: 22
random_state: 42
reg_alpha: 0.0
reg_lambda: 0.9
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: CE: Detrended Diffusion
objective: reg:squarederror
enable_categorical: False
max_depth: 10
min_child_weight: 29
missing: nan
n_estimators: 120
random_state: 42
reg_alpha: 0.2
reg_lambda: 0.2
eta: 0.2

LightGMB: CE: Original Drift
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.02
max_depth: 3
min_child_samples: 76
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 465
num_leaves: 78
random_state: 42
reg_alpha: 0.1
reg_lambda: 0.9
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: CE: Original Drift
objective: reg:squarederror
enable_categorical: False
max_depth: 4
min_child_weight: 28
missing: nan
n_estimators: 762
random_state: 42
reg_alpha: 0.2
reg_lambda: 0.1
eta: 0.02

LightGMB: CE: Original Diffusion
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.13
max_depth: 11
min_child_samples: 48
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 550
num_leaves: 22
random_state: 42
reg_alpha: 0.0
reg_lambda: 0.9
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

XGBoost: CE: Original Diffusion
objective: reg:squarederror
enable_categorical: False
max_depth: 10
min_child_weight: 29
missing: nan
n_estimators: 120
random_state: 42
reg_alpha: 0.2
reg_lambda: 0.2
eta: 0.2


results/ml_v2/model_parameters_v2.txt

AUS: Detrended Drift (Gradient Boosted Tree, Mean Squared Error, LightGMB)
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.01
max_depth: 10
min_child_samples: 39
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 610
num_leaves: 25
random_state: 42
reg_alpha: 0.4
reg_lambda: 0.1
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

AUS: Detrended Drift (Gradient Boosted Tree, Squared Error, XGBoost)
objective: reg:squarederror
enable_categorical: False
max_depth: 6
min_child_weight: 11
missing: nan
n_estimators: 950
random_state: 42
reg_alpha: 0.5
reg_lambda: 0.6
eta: 0.09

AUS: Detrended Drift (Gradient Boosted Tree, Absolute Error, XGBoost)
objective: reg:absoluteerror
enable_categorical: False
max_depth: 9
min_child_weight: 13
missing: nan
n_estimators: 530
random_state: 42
reg_alpha: 0.06
reg_lambda: 0.6
eta: 0.01

AUS: Detrended Drift (Random Forest, LightGBM)
boosting_type: rf
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.1
max_depth: 8
min_child_samples: 31
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 918
num_leaves: 51
random_state: 42
reg_alpha: 0.015
reg_lambda: 0.3
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0
bagging_freq: 3
bagging_fraction: 0.5

AUS: Detrended Drift (Multi Layer Perceptron)
activation: relu
alpha: 0.01
batch_size: auto
beta_1: 0.9
beta_2: 0.999
early_stopping: False
epsilon: 1e-08
hidden_layer_sizes: (50, 50, 50, 50)
learning_rate: constant
learning_rate_init: 0.001
max_fun: 15000
max_iter: 100
momentum: 0.9
n_iter_no_change: 10
nesterovs_momentum: True
power_t: 0.5
random_state: 42
shuffle: True
solver: adam
tol: 0.0001
validation_fraction: 0.1
verbose: False
warm_start: False

AUS: Detrended Diffusion (Gradient Boosted Tree, Mean Squared Error, LightGMB)
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.14
max_depth: 11
min_child_samples: 48
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 550
num_leaves: 22
random_state: 42
reg_alpha: 0.0004
reg_lambda: 0.9
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

AUS: Detrended Diffusion (Gradient Boosted Tree, Squared Error, XGBoost)
objective: reg:squarederror
enable_categorical: False
max_depth: 6
min_child_weight: 24
missing: nan
n_estimators: 510
random_state: 42
reg_alpha: 0.0
reg_lambda: 0.3
eta: 0.42

AUS: Detrended Diffusion (Gradient Boosted Tree, Absolute Error, XGBoost)
objective: reg:absoluteerror
enable_categorical: False
max_depth: 11
min_child_weight: 17
missing: nan
n_estimators: 624
random_state: 42
reg_alpha: 0.6
reg_lambda: 0.7
eta: 0.05

AUS: Detrended Diffusion (Random Forest, LightGBM)
boosting_type: rf
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.1
max_depth: 6
min_child_samples: 76
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 737
num_leaves: 21
random_state: 42
reg_alpha: 0.001
reg_lambda: 0.3
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0
bagging_freq: 6
bagging_fraction: 0.3

AUS: Detrended Diffusion (Multi Layer Perceptron)
activation: identity
alpha: 0.0001
batch_size: auto
beta_1: 0.9
beta_2: 0.999
early_stopping: False
epsilon: 1e-08
hidden_layer_sizes: (10, 50, 50, 10)
learning_rate: constant
learning_rate_init: 0.01
max_fun: 15000
max_iter: 100
momentum: 0.9
n_iter_no_change: 10
nesterovs_momentum: True
power_t: 0.5
random_state: 42
shuffle: True
solver: adam
tol: 0.0001
validation_fraction: 0.1
verbose: False
warm_start: False

CE: Detrended Drift (Gradient Boosted Tree, Mean Squared Error, LightGMB)
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.03
max_depth: 5
min_child_samples: 75
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 873
num_leaves: 14
random_state: 42
reg_alpha: 0.1
reg_lambda: 1.0
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

CE: Detrended Drift (Gradient Boosted Tree, Squared Error, XGBoost)
objective: reg:squarederror
enable_categorical: False
max_depth: 7
min_child_weight: 20
missing: nan
n_estimators: 381
random_state: 42
reg_alpha: 0.0
reg_lambda: 0.1
eta: 0.04

CE: Detrended Drift (Gradient Boosted Tree, Absolute Error, XGBoost)
objective: reg:absoluteerror
enable_categorical: False
max_depth: 10
min_child_weight: 16
missing: nan
n_estimators: 988
random_state: 42
reg_alpha: 1.0
reg_lambda: 0.8
eta: 0.02

CE: Detrended Drift (Random Forest, LightGBM)
boosting_type: rf
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.1
max_depth: 10
min_child_samples: 52
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 669
num_leaves: 73
random_state: 42
reg_alpha: 0.007
reg_lambda: 0.5
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0
bagging_freq: 6
bagging_fraction: 0.4

CE: Detrended Drift (Multi Layer Perceptron)
activation: relu
alpha: 0.0005
batch_size: auto
beta_1: 0.9
beta_2: 0.999
early_stopping: False
epsilon: 1e-08
hidden_layer_sizes: (20, 100, 100, 20)
learning_rate: constant
learning_rate_init: 0.005
max_fun: 15000
max_iter: 100
momentum: 0.9
n_iter_no_change: 10
nesterovs_momentum: True
power_t: 0.5
random_state: 42
shuffle: True
solver: adam
tol: 0.0001
validation_fraction: 0.1
verbose: False
warm_start: False

CE: Detrended Diffusion (Gradient Boosted Tree, Mean Squared Error, LightGMB)
boosting_type: gbdt
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.14
max_depth: 11
min_child_samples: 48
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 550
num_leaves: 22
random_state: 42
reg_alpha: 0.0
reg_lambda: 0.9
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0

CE: Detrended Diffusion (Gradient Boosted Tree, Squared Error, XGBoost)
objective: reg:squarederror
enable_categorical: False
max_depth: 10
min_child_weight: 29
missing: nan
n_estimators: 120
random_state: 42
reg_alpha: 0.2
reg_lambda: 0.2
eta: 0.2

CE: Detrended Diffusion (Gradient Boosted Tree, Absolute Error, XGBoost)
objective: reg:absoluteerror
enable_categorical: False
max_depth: 11
min_child_weight: 14
missing: nan
n_estimators: 900
random_state: 42
reg_alpha: 0.6
reg_lambda: 0.9
eta: 0.07

CE: Detrended Diffusion (Random Forest, LightGBM)
boosting_type: rf
colsample_bytree: 1.0
importance_type: split
learning_rate: 0.1
max_depth: 6
min_child_samples: 76
min_child_weight: 0.001
min_split_gain: 0.0
n_estimators: 737
num_leaves: 21
random_state: 42
reg_alpha: 0.001
reg_lambda: 0.3
subsample: 1.0
subsample_for_bin: 200000
subsample_freq: 0
bagging_freq: 6
bagging_fraction: 0.3

CE: Detrended Diffusion (Multi Layer Perceptron)
activation: identity
alpha: 0.0001
batch_size: auto
beta_1: 0.9
beta_2: 0.999
early_stopping: False
epsilon: 1e-08
hidden_layer_sizes: (10, 100, 10)
learning_rate: constant
learning_rate_init: 0.005
max_fun: 15000
max_iter: 100
momentum: 0.9
n_iter_no_change: 10
nesterovs_momentum: True
power_t: 0.5
random_state: 42
shuffle: True
solver: adam
tol: 0.0001
validation_fraction: 0.1
verbose: False
warm_start: False


utils/settings.py
# This file holds the settings for customizing each script. As some settings apply to multiple scripts,
# they are stored here instead

import matplotlib as mpl
import pandas as pd

########################
### GENERAL SETTINGS ###
########################

settings = {
    'AUS': {
        'name': 'Australia',           # full name of the area
        'reference frequency': 50,     # reference frequency of the area
        'values per hour': 900,        # number of values of the time series for a full hour
        'detrend sigma': 15,           # gaussian filter sigma used for detrending
        'outlier percent': 1           # percent of outliers that is filtered out
    },
    'CE': {
        'name': 'Continental Europe',  # full name of the area
        'reference frequency': 50,     # reference frequency of the area
        'values per hour': 3600,       # number of values of the time series for a full hour
        'detrend sigma': 60,           # gaussian filter sigma used for detrending
        'outlier percent': 2           # percent of outliers that is filtered out
    }
}

# display all columns
pd.set_option('display.max_columns', None)
pd.set_option('display.width', 2000)


########################
##### KRAMERS MOYAL ####
########################

km = {
    'AUS': {
        'drift bw': 0.05,      # Drift bandwidth
        'diffusion bw': 0.05,  # Diffusion bandwidth
        'delta_t': 4           # Resolution of time series in seconds
    },
    'CE': {
        'drift bw': 0.05,      # Drift bandwidth
        'diffusion bw': 0.05,  # Diffusion bandwidth
        'delta_t': 1           # Resolution of time series in seconds
    }
}


########################
### MACHINE LEARNING ###
########################

ml = {
    'random noise': False,                         # Add random noise feature to models
    'knockout': False,                             # Knockout the most important feature (config below)
    'test size': 0.2,                              # Test size for Machine Learning models
    'random search gbt_lgb': False,                # Perform random search: Gradient Boosted Trees, LightGBM
    'random search gbt_xgb_squarederror': False,   # Perform random search: Gradient Boosted Trees, XGBoost, Squared Error
    'random search gbt_xgb_absoluteerror': False,  # Perform random search: Gradient Boosted Trees, XGBoost, Absolute Error
    'random search rf_lgb': False,                 # Perform random search: Random Forest, LightGBM
    'grid search mlp': False,                      # Perform grid search: Multi Layer Perceptron
    'random search iterations': 1000,              # number of iterations for the random search
}

top_features = {
    'AUS': {
        'drift': {
            'gbt_lgb': 'hour_sin',
            'gbt_xgb_squarederror': 'hour_sin',
            'gbt_xgb_absoluteerror': 'hour_sin',
            'rf_lgb': 'hour_sin',
            'mlp': 'hour_sin'  # feature importance not implemented but most common upon other models chosen
        },
        'diffusion': {
            'gbt_lgb': 'hour_cos',
            'gbt_xgb_squarederror': 'hour_cos',
            'gbt_xgb_absoluteerror': 'hour_cos',
            'rf_lgb': 'hour_cos',
            'mlp': 'hour_cos'  # feature importance not implemented but most common upon other models chosen
        }
    },
    'CE': {
        'drift': {
            'gbt_lgb': 'load',
            'gbt_xgb_squarederror': 'load',
            'gbt_xgb_absoluteerror': 'total_gen',
            'rf_lgb': 'total_gen',
            'mlp': 'total_gen'  # feature importance not implemented but most common upon other models chosen
        },
        'diffusion': {
            'gbt_lgb': 'gen_nuclear',
            'gbt_xgb_squarederror': 'hour_cos',
            'gbt_xgb_absoluteerror': 'gen_nuclear',
            'rf_lgb': 'gen_nuclear',
            'mlp': 'gen_nuclear'  # feature importance not implemented but most common upon other models chosen
        }
    }
}


########################
####### PLOTTING #######
########################

# this colormap has 20 different colors, grouped in blocks of 2 that have the same base color
colormap = mpl.colormaps['tab20']

plotting = {
    'font':  'Times New Roman',
    'fontsize': 16,
    'subplot title size': 18,
    'title size': 20,
    'color AUS': {
        'frequency': colormap(14),               # grey
        'detrended drift': colormap(4),          # green
        'detrended diffusion': colormap(2),      # orange
        'original drift': colormap(5),           # light green
        'original diffusion': colormap(3),       # light orange
    },
    'color CE': {
        'frequency': colormap(10),               # brown
        'detrended drift': colormap(0),          # blue
        'detrended diffusion': colormap(6),      # red
        'original drift': colormap(1),           # light blue
        'original diffusion': colormap(7),       # light red
    },
    'color_gbt_lgb': colormap(12),               # pink
    'color_gbt_xgb_squarederror': colormap(8),   # purple
    'color_gbt_xgb_absoluteerror': colormap(9),  # light purple
    'color_rf_lgb': colormap(18),                # cyan
    'color_mlp': colormap(19),                   # light cyan

    # specific plotting settings for script 02
    'plotting type': 'average and frequency',    # possible options 'average', 'sequential', 'average and frequency'
    #                 average:                     avg over all time intervals of the given length
    #                 sequential:                  plot the intervals on top of each other without averaging
    #                 average and frequency:       plot the avg of the drift, diffusion and avg of the frequency
    'sequential limit': 2,                       # max number of sequential intervals plotted
    'rolling average': True                      # adds the rolling average to the plot
}

# set font
mpl.rcParams['font.family'] = plotting['font']
mpl.rc('legend', fontsize=plotting['fontsize'])

